client/core/core.go

```diff
@@ -121,6 +121,9 @@ func (dc *dexConnection) marketConfig(mktID string) *msgjson.Market {
 // marketMap creates a map of this DEX's *Market keyed by name/ID,
 // [base]_[quote].
 func (dc *dexConnection) marketMap() map[string]*Market {
+	if atomic.LoadUint32(&dc.connected) == 0 {
+		return nil
```

Comment:
What's the motivation for this change?  Was there an existing bug?  It's not clear to me this is going to be a benign change in all cases.

Reply:
Right now when calling the `User` method on core, it calls the method  `exchangeMap`, which calls the method `marketMap` from the connection. 

As I am returning the connection now, when it fails to connect, I need this check, otherwise, `dc.cfg` will be nil, and break things.

Reply:
If what you are really after is `dc.cfg` being nil or not, I would prefer checking that.